jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/EcsExecUtils.kt

```diff
@@ -29,24 +28,13 @@ class EcsExecUtils(private val project: Project) : CoroutineScope by Application
         }
     }
 
-    suspend fun checkServiceState(service: Service): Boolean {
-        return try {
-            do {
-                delay(5000)
-                val response = checkServiceCompletion(service, project)
-            } while (response.deployments().first().rolloutStateAsString() != "COMPLETED")
-            true
-        } catch (e: Exception) {
-            false
-        }
+    suspend fun checkServiceState(project: Project, service: Service): Boolean {
+        delay(5000)
```

Comment:
Why do we need the explict delay?

Reply:
If we call describe-services immediately, it returns the previous stable state. There is some latency before the updated service progress is displayed

Reply:
ah, funsies. Thanks, can you add a comment on why please